libs/ledger-live-common/src/families/cosmos/api/Cosmos.ts

```diff
@@ -72,16 +76,26 @@ export class CosmosAPI {
         url: `${this.defaultEndpoint}/cosmos/auth/${this.version}/accounts/${address}`,
       });
 
-      if (data.account.account_number) {
-        response.accountNumber = parseInt(data.account.account_number);
+      const srcAccount =
+        data.account.base_account != null ? data.account.base_account : data.account;
+
+      if (srcAccount.account_number) {
+        accountData.accountNumber = parseInt(srcAccount.account_number);
+      }
+
+      if (srcAccount.sequence) {
+        accountData.sequence = parseInt(srcAccount.sequence);
       }
 
-      if (data.account.sequence) {
-        response.sequence = parseInt(data.account.sequence);
+      if (srcAccount.pub_key) {
+        accountData.pubKey = srcAccount.pub_key.key;
+        accountData.pubKeyType = srcAccount.pub_key["@type"];
       }
-      // eslint-disable-next-line no-empty
-    } catch (e) {}
-    return response;
+    } catch (e) {
+      console.log("Could not fetch account info, using default values instead");
```

Comment:
Is this really what we want? Just asking, because sometimes using default values could actually cause bugs and just throwing an error is cleaner.
Let's review how the returned fields are used and what the impact is with default values, to assess if they are ok or not.

(Also, `console.log` being only available in dev env, for logs that we actually want to add in prod `log` should be used instead)

Reply:
To be honest, I don't know if it's necessary, I already changed a lot so I didn't want to do any breaking change here...

I agree throwing would be cleaner

Will add log instead while we discuss this

Reply:
Confirmed : https://injective-api.polkachu.com/cosmos/auth/v1beta1/accounts/inj1ddscefl2mnv5sfnduesz7tu0x3fdrttven4f9m

It is needed for a pristine account, we need a 0 sequence in that case else we are stuck

Reply:
:ok_hand: Thanks for checking :raised_hands: